data/metric/metrics.go

```diff
@@ -85,11 +89,15 @@ func isNumeric(value interface{}) bool {
 	return err == nil
 }
 
+func castToNumeric(value interface{}) (float64, error) {
```

Comment:
I have just one picky comment, the rest is fine :) 
Having this function `castToNumeric`, is it still needed to have `isNumeric` https://github.com/newrelic/infra-integrations-sdk/pull/95/files#diff-83416e4d39b0688a4bec5f4a2c076486R87? Shouldn't we just get rid of `isNumeric`?